checkov/cloudformation/runner.py

```diff
@@ -1,48 +1,58 @@
+from checkov.common.checks_infra.registry import get_graph_checks_registry
+from checkov.common.graph.db_connectors.networkx.networkx_db_connector import NetworkxConnector
+from checkov.common.graph.graph_builder import CustomAttributes
+from checkov.common.output.graph_record import GraphRecord
 from checkov.common.output.record import Record
-from checkov.common.output.report import Report
+from checkov.common.output.report import Report, merge_reports
 from checkov.common.runners.base_runner import BaseRunner
 from checkov.runner_filter import RunnerFilter
 
 
 class Runner(BaseRunner):
     check_type = "cloudformation"
 
+    def __init__(self, db_connector=NetworkxConnector(),
+                 source="CloudFormation",
+                 graph_class=CloudformationLocalGraph,
+                 graph_manager=None,
+                 external_registries=None):
+        self.external_registries = [] if external_registries is None else external_registries
+        self.graph_class = graph_class
+        self.graph_manager = graph_manager if graph_manager is not None else CloudformationGraphManager(source=source,
+                                                                                                        db_connector=db_connector)
+        self.definitions_raw = {}
+        self.graph_registry = get_graph_checks_registry(self.check_type)
+
     def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=RunnerFilter(), collect_skip_comments=True):
         report = Report(self.check_type)
-        definitions = {}
-        definitions_raw = {}
-        if external_checks_dir:
-            for directory in external_checks_dir:
-                cfn_registry.load_external_checks(directory)
 
-        if files:
-            for file in files:
-                (definitions[file], definitions_raw[file]) = parse(file)
+        if self.context is None or self.definitions is None or self.breadcrumbs is None:
 
-        if root_folder:
-            definitions, definitions_raw = get_folder_definitions(root_folder, runner_filter.excluded_paths)
+            self.definitions, self.definitions_raw = create_definitions(root_folder, files, runner_filter)
+            if external_checks_dir:
+                for directory in external_checks_dir:
+                    cfn_registry.load_external_checks(directory)
 
-        # Filter out empty files that have not been parsed successfully, and filter out non-CF template files
-        definitions = {k: v for k, v in definitions.items() if v and isinstance(v, dict_node) and v.__contains__("Resources") and isinstance(v["Resources"], dict_node)}
-        definitions_raw = {k: v for k, v in definitions_raw.items() if k in definitions.keys()}
+        self.context = build_definitions_context(self.definitions, self.definitions_raw, root_folder)
 
-        for cf_file in definitions.keys():
+        for cf_file in self.definitions.keys():
```

Comment:
```suggestion
        for cf_file, definition in self.definitions.items():
```
then you don't need to use `self.definitions[cf_file]` further on.